interfaces/Portalicious/src/app/auth.guard.ts

```diff
@@ -18,3 +19,30 @@ export const authGuard: CanActivateFn = (_, state) => {
     queryParamsHandling: 'merge',
   });
 };
+
+export const projectPermissionsGuard: (
+  permission: PermissionEnum,
+) => CanActivateFn = (permission: PermissionEnum) => {
+  return function projectPermissionsCanActivateFn(
+    route: ActivatedRouteSnapshot,
+  ) {
+    const authService = inject(AuthService);
+
+    const projectId = route.params.projectId as number;
+
+    if (isNaN(projectId)) {
+      throw new Error('projectId is not a number');
+    }
+
+    if (authService.hasPermission(projectId, permission)) {
```

Comment:
There is no "needs multiple permissions"-functionality yet, I think?

Reply:
Tbh, I don't know 😬